io.openems.edge.evcs.spelsberg.smart/src/io/openems/edge/evcs/spelsberg/smart/EvcsSpelsbergSmartImpl.java

```diff
@@ -0,0 +1,343 @@
+	}
+
+	@Activate
+	private void activate(ComponentContext context, Config config) throws OpenemsNamedException {
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+
+		/*
+		 * Calculates the maximum and minimum hardware power dynamically by listening on
+		 * the fixed hardware limits used for charging
+		 */
+		Evcs.addCalculatePowerLimitListeners(this);
+		applyConfig(config);
+	}
+
+	@Modified
+	private void modified(ComponentContext context, Config config) throws OpenemsNamedException {
+		if (super.modified(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		applyConfig(config);
+	}
+
+	@Override
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+
+		var modbusProtocol = new ModbusProtocol(this,
+				new FC3ReadRegistersTask(1000, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_POINT_STATE, new UnsignedWordElement(1000)),
+						new DummyRegisterElement(1001),
+						m(EvcsSpelsbergSmart.ChannelId.EVSE_STATE, new UnsignedWordElement(1002))),
+
+				new FC3ReadRegistersTask(1004, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CABLE_STATE, new UnsignedWordElement(1004))),
+
+				new FC3ReadRegistersTask(1008, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L1, new UnsignedWordElement(1008))),
+
+				new FC3ReadRegistersTask(1010, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L2, new UnsignedWordElement(1010))),
+
+				new FC3ReadRegistersTask(1012, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CURRENT_L3, new UnsignedWordElement(1012))),
+
+				new FC3ReadRegistersTask(1020, Priority.HIGH,
+						m(Evcs.ChannelId.CHARGE_POWER, new UnsignedDoublewordElement(1020))),
+
+				new FC3ReadRegistersTask(1020, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_TOTAL, new UnsignedDoublewordElement(1020))),
+
+				new FC3ReadRegistersTask(1024, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L1, new UnsignedDoublewordElement(1024))),
+
+				new FC3ReadRegistersTask(1028, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L2, new UnsignedDoublewordElement(1028))),
+
+				new FC3ReadRegistersTask(1032, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.POWER_L3, new UnsignedDoublewordElement(1032))),
+
+				new FC3ReadRegistersTask(1036, Priority.LOW,
+						m(Evcs.ChannelId.ACTIVE_CONSUMPTION_ENERGY, new UnsignedDoublewordElement(1036))),
+
+				new FC3ReadRegistersTask(1100, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.MAX_HARDWARE_CURRENT, new UnsignedWordElement(1100))),
+
+				new FC3ReadRegistersTask(1102, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.MIN_HARDWARE_CURRENT, new UnsignedWordElement(1102))),
+
```

Comment:
Try combining registers in a Task by using dummyRegisters in between if they are next to each other with only a few reserved or empty registers.

Reply:
I have combined the registeres in a single task and added dummyRegisters where possible.